### ISSUE TITLE
Problem with assert referenced columns are present in datasource

### ISSUE THREAD

**Issue by lilila** [apache/incubator-superset#11078]
@villebro , I have an issue with #10451 

It breaks the following type of jinja-based filters:

``` 
{% set min_tracking  = filter_values('min_sem_tracking', -9 )[0] %} 
{% set max_tracking = filter_values('max_sem_tracking', 2 )[0]  %} 


SELECT 
...
WHERE 
sem_tracking > {min_tracking} AND sem_tracking < {max_tracking} 
```

**Comment by villebro** [comment]
Thanks for posting this issue @lilila . Can you post some more details on how to reproduce the error?

**Comment by lilila** [comment]
Hi Villebro, 
If I want to add a min / max filter on a field (since the filter box does not allow this) I usually create a SQL Lab view as follows :

`SELECT 
-5 as min_sem_tracking, 
1 as max_sem_tracking
`
Then I create a filter  box with the 2 fields created in this "fake" table : 
<img width="1089" alt="Screenshot 2020-09-28 at 06 35 50" src="https://user-images.githubusercontent.com/607430/94391078-dbb6c100-0154-11eb-9ad5-7b17202a4bc8.png">

After adding the filter box in the dashboard I use the user input value for min and max value to create more "complex"  filters on another table.  As explain above I collect user inputs using `filter_values` 

```
{% set min_tracking  = filter_values('min_sem_tracking', -9 )[0] %} 
{% set max_tracking = filter_values('max_sem_tracking', 2 )[0]  %} 
```

Then I use the collected values to create filters on another field in another table : 

```
SELECT * FROM my_table 
WHERE 
sem_tracking > {min_tracking} AND sem_tracking < {max_tracking} 
```

With the new restriction, I receive an error message saying `min_sem_tracking` and `max_sem_tracking` are not in the datasource.

**Comment by villebro** [comment]
I was able to reproduce, thanks @lilila and @bkyryliuk for helping track this down. Working on a fix now.

**Comment by lilila** [comment]
Hi @villebro , I hope you are doing well. 
I tried the new release (1.0.rc1) and I have the impression the problem is back. All "artificial filters " (such as min / max values) generate warnings. In the end I end up with a dashboard looking like this : 
<img width="1723" alt="Screenshot 2021-01-13 at 10 18 45" src="https://user-images.githubusercontent.com/607430/104433161-128cd780-558a-11eb-8a04-3d1f66e08dda.png">

**Comment by villebro** [comment]
@lilila is the behavior consistent? There is a problem that causes chart loading errors on `1.0.0rc1` that should be fixed by this PR: #12489
